### PR TITLE
docs(dump): added openapi doc

### LINKED ISSUE #2: Design API spec for the service
I would like to have a design for the API before we move on to implement the service.

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Closes #2

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -1,46 +1,100 @@
...
     name: MIT
     url: https://opensource.org/licenses/MIT
 servers:
   - url: http://localhost:8080
 paths:
-  /resourceName:
+  /dumps:
     get:
-      operationId: getResourceName
+      operationId: getDumpsByQuery
```
Comment (CptSchnitz) [inline openapi3.yaml:14]: No need for by query

File: openapi3.yaml [hunk 2]
```diff
@@ -1,46 +1,100 @@
...
-      summary: gets the resource
+        - dump
+      summary: Get dumps by query
+      parameters:
+        - in: query
+          name: limit
+          schema:
+            type: integer
+            default: 10
+            minimum: 0
```
Comment (CptSchnitz) [inline openapi3.yaml:24]: should be minimum 1

File: openapi3.yaml [hunk 3]
```diff
@@ -1,46 +1,100 @@
...
-        - resourceName
-      summary: creates a new record of type resource
+        - dump
+      summary: Get a dump by id
+      parameters:
+        - in: path
+          name: dumpId
+          schema:
+            type: string
+            format: uuid
```
Comment (CptSchnitz) [inline openapi3.yaml:70]: should be a schema

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: openapi3.yaml (hunk 3 above)
Comment (melancholiai) [reply openapi3.yaml]: done.

File: openapi3.yaml (hunk 2 above)
Comment (melancholiai) [reply openapi3.yaml]: done.

File: openapi3.yaml (hunk 1 above)
Comment (melancholiai) [reply openapi3.yaml]: done.

**Review by syncush — CHANGES_REQUESTED** [review]

File: openapi3.yaml [hunk 4]
```diff
@@ -1,46 +1,100 @@
...
+                $ref: '#/components/schemas/dump'
         400:
-          description: Bad Request
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/error'
+          $ref: '#/components/responses/BadRequest'
+        404:
+          $ref: '#/components/responses/NotFound'
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply openapi3.yaml:80]: done.

File: openapi3.yaml [hunk 5]
```diff
@@ -1,46 +1,100 @@
...
-                $ref: '#/components/schemas/resource'
-    post:
-      operationId: createResource
+                type: array
+                items:
+                  $ref: '#/components/schemas/dump'
+        400:
+          $ref: '#/components/responses/BadRequest'
+        404:
+          $ref: '#/components/responses/NotFound'
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply openapi3.yaml]: done.

**Review by syncush — APPROVED** [review]

**Pull request merged**